stdlib/LinearAlgebra/test/uniformscaling.jl

```diff
@@ -400,6 +400,11 @@ end
     @test 0denseI != 2I != 0denseI # test generic path / inequality on diag
     @test alltwos != 2I != alltwos # test generic path / inequality off diag
     @test rdenseI !=  I != rdenseI # test square matrix check
+
+    # isequal
+    @test !isequal(I, I(3))
+    @test !isequal(I(1), I)
+    @test !isequal([1], I)
```

Comment:
I first thought your approach would cause an ambiguity error for `isequal(I, I)`, but fortunately it doesn't because `UniformScaling` doesn't subtype `AbstractMatrix`. Nethertheless, would you mind adding a test for `isequal(I, 1I)` just to be sure this doesn't break in the future?

Reply:
Of course, added now.